Task 1

Damovetsky opened the pull request.

Commit:
Realized the general design of my_tasks_screen

Commit:
Added a tasks provider and some UI elements

Commit:
Designed task detail screen

Commit:
Realized swipe to check/delete logic

Commit:
Completed tasks counter && hide tasks logic

Commit:
Minor fixes in the tile UI

Commit:
Added navigation and task editing/creating logic

Commit:
Added calendar logic

Commit:
Finished priority logic

Commit:
Added a logo for the app

Commit:
Added logging and fixed some stuff

Commit:
Changed README file

rn7cvj commented:
Привет 🐳🐳🐳 я ревьюю твой код на ШМР.
ReadMe - 4 / 4
Верста - 10 / 10 - есть проблемы с appBar issue уже кинул + нету кнопки при прокручивании в appBar( проблемы сильно не критичны, не думаю что стоит срезать половину баллов, но пофикси) 
Свайп  - 3/3
Показ выполненный дел с логикой - 0 / 3 - тут сорян у тебя id в Task почему-то это строка, + сбивается приоритет при редактировании issue тоже кинул, можно было и оставить 1 балл, но сниму за косяк с appBar
Логирование - 1 / 2 - у тебя в каждом виджете свой экземпляр логера, нету централизованной настройки вывода 
Иконка - 0 / 4 - нам говорили что нельзя юзать пакеты, это делается в пару кликов в Android Studio, разберись 😊😊😊
Итого - 18 / 26

TheLastFlame commented:
Привет ^^

Ну, что я могу сказать
Выход в свет... не задался
Всё грустно.

Я очень не хочу этого делать, но:

Иконка -- 0/4, нам запретили использовать flutter_launcher_icons.

Вёрстка -- 5/10, аппбару реально очень плохо. И ладно бы оно просто кривым было, но оно влияет на функциональность (отсутствует переключение видимости завершенных при свёрнутом аппбаре). Хотелось бы снять поменьше, но у нас по правилам 1/2 за критерий при критичных проблемах.

И в целом его реализация крайне костыльная.
Немножко советов:
1. Не используй title для этого. Ты просто усложнил себе жизнь. У FlexibleSpacerBar есть background, который сам скрывается при прокрутке.
2. Ты ставил падинги на ощупь, для своего девайса. Ещё зачем-то использовал SafeArea... это всё не помогло. На моём устройстве аппбар очень криво обрезан. А при некоторых соотношениях сторон вообще часть скрывающихся элементов остаётся видна.
3. Вместо SingleChildScrollView с отключенной прокруткой можно использовать OverflowBox. (Но не здесь, просто на будущее говорю)

![image](https://github.com/Damovetsky/totodo/assets/131446187/bbc9427d-6a15-49aa-af8b-10a933ae14aa)

Показ выполненных дел с логикой -- 1,5/3, есть критичный баг со сбросом приоритета при редактировании таска.
(Открываешь таск, там указан приоритет "Важное". Нажимаешь "сохранить". Внезапно приоритет меняется на нормальный. Смотрю код. Судя по всему, ты создаёшь новый объект таска. И забываешь присвоить ему приоритет из полученного в аргументах таска, хоть и отображаешь его) 

По коду, структуре... В целом, понравилось. Классно, круто, прям огонь. Удивило, что ты вообще не использовал менеджеры состояний. Только стандартные sf виджеты и ChangeNotifier. Не особо применимо в более крупной разработке, но иметь такой опыт весьма полезно, ты хорош.

Итог: 16/26
(15.5, но округляем в твою пользу)

Самый главный совет:
Читай чаты. Мы многое обсуждаем по реализации. Там можно задавать свои вопросы и т.д.

rn7cvj commented:
> Привет ^^
> 
> Ну, что я могу сказать Выход в свет... не задался Всё грустно.
> 
> Я очень не хочу этого делать, но:
> 
> Иконка -- 0/4, нам запретили использовать flutter_launcher_icons.
> 
> Вёрстка -- 5/10, аппбару реально очень плохо. И ладно бы оно просто кривым было, но оно влияет на функциональность (отсутствует переключение видимости завершенных при свёрнутом аппбаре). Хотелось бы снять поменьше, но у нас по правилам 1/2 за критерий при критичных проблемах.
> 
> И в целом его реализация крайне костыльная. Немножко советов:
> 
> 1. Не используй title для этого. Ты просто усложнил себе жизнь. У FlexibleSpacerBar есть background, который сам скрывается при прокрутке.
> 2. Ты ставил падинги на ощупь, для своего девайса. Ещё зачем-то использовал SafeArea... это всё не помогло. На моём устройстве аппбар очень криво обрезан. А при некоторых соотношениях сторон вообще часть скрывающихся элементов остаётся видна.
> 3. Вместо SingleChildScrollView с отключенной прокруткой можно использовать OverflowBox. (Но не здесь, просто на будущее говорю)
> 
> ![image](https://user-images.githubusercontent.com/131446187/246659202-bbc9427d-6a15-49aa-af8b-10a933ae14aa.png)
> 
> Показ выполненных дел с логикой -- 1,5/3, есть критичный баг со сбросом приоритета при редактировании таска. (Открываешь таск, там указан приоритет "Важное". Нажимаешь "сохранить". Внезапно приоритет меняется на нормальный. Глянул код. Судя по всему, ты создаёшь новый объект таска. И забываешь присвоить ему приоритет из полученного в аргументах таска, хоть и отображаешь его)
> 
> Пробежался по коду, структуре. В целом, понравилось. Классно, круто, прям огонь. Удивило, что ты вообще не использовал менеджеры состояний. Только стандартные sf виджеты и ChangeNotifier. Не особо применимо в более крупной разработке, но иметь такой опыт весьма полезно, ты хорош.
> 
> Итог: 16/26 (15.5, но округляем в твою пользу)
> 
> Самый главный совет: Читай чаты. Мы многое обсуждаем по реализации. Там можно задавать свои вопросы и т.д.

Пиздец ты пассивно агрессивный

Damovetsky commented:
> Привет 🐳🐳🐳 я ревьюю твой код на ШМР. ReadMe - 4 / 4 Верста - 10 / 10 - есть проблемы с appBar issue уже кинул + нету кнопки при прокручивании в appBar( проблемы сильно не критичны, не думаю что стоит срезать половину баллов, но пофикси) Свайп - 3/3 Показ выполненный дел с логикой - 0 / 3 - тут сорян у тебя id в Task почему-то это строка, + сбивается приоритет при редактировании issue тоже кинул, можно было и оставить 1 балл, но сниму за косяк с appBar Логирование - 1 / 2 - у тебя в каждом виджете свой экземпляр логера, нету централизованной настройки вывода Иконка - 0 / 4 - нам говорили что нельзя юзать пакеты, это делается в пару кликов в Android Studio, разберись 😊😊😊 Итого - 18 / 26

Привет, в первую очередь спасибо, что подробно разобрал мою работу и оставил комментарии к проверке.
Согласен с тем, что appbar я пока ещё не довёл до надлежащего вида. Уже получил совет, как его можно исправить, постараюсь реализовать. 
Всё же не совсем согласен с оценкой по "показу выполненный дел с логикой". Функционал ведь полностью работает, ничего критично не ломается. Про баг с приоритетом уже знаю, вчера его отловил, но там фикс в одну строчку. Не понимаю, что плохого в id для task строкой, раньше работал с Firebase, там вроде как строкой объекты идентифицируются. Но дело твоё. Если считаешь, что 0 справедливая оценка, ставь.
Про иконку я ещё у ментора спрошу. Ты прав, в беседе писали, что flutter_launcher_icons нельзя использовать, мимо меня это прошло. А я подумал, что для установки иконки для Android и IOS надо в нативные файлы лезть, а раз лекции по нативу у нас ещё на было, то и ожидать этого от нас не будут. А раз иконку я сам разрабатывал, то работа по пункту всё равно проведена. 
В любом случае, спасибо за оценивание. Желаю тебе успехов в дальнейших работах!

rn7cvj commented:
> > Привет 🐳🐳🐳 я ревьюю твой код на ШМР. ReadMe - 4 / 4 Верста - 10 / 10 - есть проблемы с appBar issue уже кинул + нету кнопки при прокручивании в appBar( проблемы сильно не критичны, не думаю что стоит срезать половину баллов, но пофикси) Свайп - 3/3 Показ выполненный дел с логикой - 0 / 3 - тут сорян у тебя id в Task почему-то это строка, + сбивается приоритет при редактировании issue тоже кинул, можно было и оставить 1 балл, но сниму за косяк с appBar Логирование - 1 / 2 - у тебя в каждом виджете свой экземпляр логера, нету централизованной настройки вывода Иконка - 0 / 4 - нам говорили что нельзя юзать пакеты, это делается в пару кликов в Android Studio, разберись 😊😊😊 Итого - 18 / 26
> 
> Привет, в первую очередь спасибо, что подробно разобрал мою работу и оставил комментарии к проверке. Согласен с тем, что appbar я пока ещё не довёл до надлежащего вида. Уже получил совет, как его можно исправить, постараюсь реализовать. Всё же не совсем согласен с оценкой по "показу выполненный дел с логикой". Функционал ведь полностью работает, ничего критично не ломается. Про баг с приоритетом уже знаю, вчера его отловил, но там фикс в одну строчку. Не понимаю, что плохого в id для task строкой, раньше работал с Firebase, там вроде как строкой объекты идентифицируются. Но дело твоё. Если считаешь, что 0 справедливая оценка, ставь. Про иконку я ещё у ментора спрошу. Ты прав, в беседе писали, что flutter_launcher_icons нельзя использовать, мимо меня это прошло. А я подумал, что для установки иконки для Android и IOS надо в нативные файлы лезть, а раз лекции по нативу у нас ещё на было, то и ожидать этого от нас не будут. А раз иконку я сам разрабатывал, то работа по пункту всё равно проведена. В любом случае, спасибо за оценивание. Желаю тебе успехов в дальнейших работах!

Привет, я уже написал что за логику стоит 0 / 3 т.к я поставил тебе за верстку 10 / 10, есть проблемы но мы можем поставить 0, 5 или 10 баллов, я не хотел снижать тебе много баллов за вёрстку, поэтому  снял баллы в менее значимых категориях, в целом в id в качестве строки нету ничего плохого, но обычно это все таки число.

Git-GA commented:
> Про баг с приоритетом уже знаю, вчера его отловил, но там фикс в одну строчку

Да большинство фиксов такие. Проблема в том, что ты уже выкатил это в так называемый прод. Но это базовый, а не дополнительный функционал. Следовательно это критичный баг логики, влияющий на функциональность.

Система правда так себе, но не мы её придумываем, к сожалению.

Можешь реально поспрашивать у ментора. Оценку менять можно, так что поправлю, если разрешат)

Damovetsky commented:
> Привет ^^
> 
> Ну, что я могу сказать Выход в свет... не задался Всё грустно.
> 
> Я очень не хочу этого делать, но:
> 
> Иконка -- 0/4, нам запретили использовать flutter_launcher_icons.
> 
> Вёрстка -- 5/10, аппбару реально очень плохо. И ладно бы оно просто кривым было, но оно влияет на функциональность (отсутствует переключение видимости завершенных при свёрнутом аппбаре). Хотелось бы снять поменьше, но у нас по правилам 1/2 за критерий при критичных проблемах.
> 
> И в целом его реализация крайне костыльная. Немножко советов:
> 
> 1. Не используй title для этого. Ты просто усложнил себе жизнь. У FlexibleSpacerBar есть background, который сам скрывается при прокрутке.
> 2. Ты ставил падинги на ощупь, для своего девайса. Ещё зачем-то использовал SafeArea... это всё не помогло. На моём устройстве аппбар очень криво обрезан. А при некоторых соотношениях сторон вообще часть скрывающихся элементов остаётся видна.
> 3. Вместо SingleChildScrollView с отключенной прокруткой можно использовать OverflowBox. (Но не здесь, просто на будущее говорю)
> 
> ![image](https://user-images.githubusercontent.com/131446187/246659202-bbc9427d-6a15-49aa-af8b-10a933ae14aa.png)
> 
> Показ выполненных дел с логикой -- 1,5/3, есть критичный баг со сбросом приоритета при редактировании таска. (Открываешь таск, там указан приоритет "Важное". Нажимаешь "сохранить". Внезапно приоритет меняется на нормальный. Смотрю код. Судя по всему, ты создаёшь новый объект таска. И забываешь присвоить ему приоритет из полученного в аргументах таска, хоть и отображаешь его)
> 
> По коду, структуре... В целом, понравилось. Классно, круто, прям огонь. Удивило, что ты вообще не использовал менеджеры состояний. Только стандартные sf виджеты и ChangeNotifier. Не особо применимо в более крупной разработке, но иметь такой опыт весьма полезно, ты хорош.
> 
> Итог: 16/26 (15.5, но округляем в твою пользу)
> 
> Самый главный совет: Читай чаты. Мы многое обсуждаем по реализации. Там можно задавать свои вопросы и т.д.

Привет, в первую очередь спасибо за подробную проверку и твои советы. 
На счёт аппбара, согласен с тобой. Честно говоря, не знал как к этой задаче подойти, поэтому сделал довольно костыльно до лучших времён. И совсем не подумал, что на другом разрешении всё может сбиться. Но всё же, считаю, что 5 баллов за это списывать довольно много. Нам писали, что по каждому экрану вёрстку можно отдельно оценивать, и тогда справедливее было бы 7,5 из 10 поставить. Но у всех нас своё понимание справедливости, так что оценивай как знаешь. А за совет по реализации большое спасибо, попробую, может получится костыли мои поправить)
Про иконку, а уже отвечал другому проверяющему. Просмотрел я, что нельзя flutter_launcher_icons использовать, но как-то больно на таком пункте столько баллов терять. Поговорю ещё с ментором, но наверное ты всё же прав. 
С багом на показе дел с логикой согласен, возможно он и правда критичный. Обидно, что на сдачу его затащил, ведь там одной строчкой фиксится. Но моё почтение за вашу внимательность, я при тестах его проморгал)
В любом случае, спасибо за проверку и крутые советы. Желаю тебе успехов в дальнейших заданиях!

TheLastFlame commented:
>  Но всё же, считаю, что 5 баллов за это списывать довольно много. Нам писали, что по каждому экрану вёрстку можно отдельно оценивать, и тогда справедливее было бы 7,5 из 10 поставить. Но у всех нас своё понимание справедливости, так что оценивай как знаешь. А за совет по реализации большое спасибо, попробую, может получится костыли мои поправить) 

Честно, не помню ничего про отдельную оценку экранов. И да, я тоже считаю, что 5 баллов очень много и достаточно обидно. И не против изменить, если договоришься с ментором ^^

> Про иконку, а уже отвечал другому проверяющему. Просмотрел я, что нельзя flutter_launcher_icons использовать, но как-то больно на таком пункте столько баллов терять. Поговорю ещё с ментором, но наверное ты всё же прав.

Насчёт иконки, у меня был длинный разговор с ментором насчёт этого пакета. Обидно, но как факт.

> С багом на показе дел с логикой согласен, возможно он и правда критичный. Обидно, что на сдачу его затащил, ведь там одной строчкой фиксится. 

Одна строка... это да, большинство проблем так и фиксятся на самом деле. Но баг действительно критичный и попал в релиз, поэтому тут как есть. Просто-напросто это недоработка не этапа разработки, а этапа тестирования.
Мной этот баг был замечен буквально на первых минутах. Просто открываю информацию о важном ивенте, ничего не меняю, нажимаю сохранить... и он внезапно перестаёт быть важным.

> Честно говоря, не знал как к этой задаче подойти

Ну и да, ещё раз, если не понимаешь, как подойти к задаче, спроси в чате) Нас там много, все учатся, и многие не против помочь. У нас есть специальный Технический чат для подобных обсуждений. И реализацию этого аппбара там несколько раз обсуждали, вплоть до конкретных инструкций.

rn7cvj commented:
> Привет 🐳🐳🐳 я ревьюю твой код на ШМР. ReadMe - 4 / 4 Верста - 10 / 10 - есть проблемы с appBar issue уже кинул + нету кнопки при прокручивании в appBar( проблемы сильно не критичны, не думаю что стоит срезать половину баллов, но пофикси) Свайп - 3/3 Показ выполненный дел с логикой - 0 / 3 - тут сорян у тебя id в Task почему-то это строка, + сбивается приоритет при редактировании issue тоже кинул, можно было и оставить 1 балл, но сниму за косяк с appBar Логирование - 1 / 2 - у тебя в каждом виджете свой экземпляр логера, нету централизованной настройки вывода Иконка - 0 / 4 - нам говорили что нельзя юзать пакеты, это делается в пару кликов в Android Studio, разберись 😊😊😊 Итого - 18 / 26

UPD: нам разрешили юзать flutter_launcher_icons, так что + 4 балла

Итого 22 / 26

Commit:
Fixes after code review
Fixed a bug with priority, tweaked appbar, but it still needs some fixes and made a centralized logger
